src/main/java/com/duckma/duckzip/DuckZip.java

```diff
@@ -160,10 +170,125 @@ public void unzip(String sourceFilePath, String destDirectoryPath,
+     * @param zipProgressUpdateCallback The interval at which the updates will be fired.
+     * @param updateInterval            A callback to receive progress updates.
+     * @throws IOException
+     */
+    public void zip(String sourcePath, String destFilePath,
+                    ZipProgressUpdateCallback zipProgressUpdateCallback,
+                    UpdateInterval updateInterval) throws IOException {
+        if (sourcePath == null) {
+            throw new IllegalArgumentException("Please provide a sourceFilePath");
+        }
+        if (destFilePath == null) {
+            throw new IllegalArgumentException("Please provide a destinationFilePath");
+        }
+
+        File zipFile = new File(destFilePath);
+        if (zipFile.exists()) {
+            zipFile.delete();
+        }
+
+        FileOutputStream zipFos = new FileOutputStream(zipFile);
+        ZipOutputStream zos = new ZipOutputStream(zipFos);
+        File root = new File(sourcePath);
+        List<File> files = getFiles(root);
+        long lastUpdateTime = System.currentTimeMillis();
+
+        for (int i = 0; i < files.size(); i++) {
+            long currentTimeMillis = currentMillisCheck.currentTimeMillis();
+            if (currentTimeMillis - lastUpdateTime > updateInterval.getMillisInterval()) {
+                zipProgressUpdateCallback.onZipProgressUpdate((float) (i+1) / files.size());
+                lastUpdateTime = currentTimeMillis;
+            }
+
+            File file = files.get(i);
+            String path = file.getPath()
+                .substring(root.isDirectory() ? root.getPath().length() : root.getParent().length());
+
+            ZipEntry zipEntry = new ZipEntry(path);
+            zos.putNextEntry(zipEntry);
+
+            FileInputStream fis = new FileInputStream(new File(file.getAbsolutePath()));
+            byte[] bytes = new byte[1024];
+            int length;
+            while ((length = fis.read(bytes)) >= 0) {
+                zos.write(bytes, 0, length);
+            }
+            fis.close();
+            zos.closeEntry();
+        }
+
+        zos.finish();
+        zos.close();
+    }
+
+    private List<File> getFiles(File file) {
+        List<File> result = new ArrayList<>();
+        if (!file.isDirectory()) {
+            result.add(file);
+        } else {
+            File[] files = file.listFiles();
+            if (files != null) {
+                for (File child : files) {
+                    result.addAll(getFiles(child));
+                }
+            }
+        }
+        return result;
+    }
+
+    public void zip(String sourceFilePath, String destDirectoryPath,
+                    ZipProgressUpdateCallback zipProgressUpdateCallback) throws IOException {
+        zip(sourceFilePath, destDirectoryPath, zipProgressUpdateCallback, UpdateInterval.UPDATE_INTERVAL_DEFAULT);
+    }
+
     public interface UnzipProgressUpdateCallback {
```

Comment:
Can we merge these 2 interface into a single one that works for both zip and unzip?